lib/sitemap-stream.ts

```diff
@@ -1,79 +1,21 @@
-export const closetag = '</urlset>';
-export interface ISitemapStreamOpts
-  extends TransformOptions,
-    Pick<ISitemapOptions, 'hostname' | 'level' | 'lastmodDateOnly'> {}
-const defaultStreamOpts: ISitemapStreamOpts = {};
-export class SitemapStream extends Transform {
-  hostname?: string;
-  level: ErrorLevel;
-  hasHeadOutput: boolean;
-  lastmodDateOnly: boolean;
-  constructor(opts = defaultStreamOpts) {
-    opts.objectMode = true;
-    super(opts);
-    this.hasHeadOutput = false;
-    this.hostname = opts.hostname;
-    this.level = opts.level || ErrorLevel.WARN;
-    this.lastmodDateOnly = opts.lastmodDateOnly || false;
-  }
-
-  _transform(
-    item: ISitemapItemOptionsLoose,
-    encoding: string,
-    callback: TransformCallback
-  ): void {
-    if (!this.hasHeadOutput) {
-      this.hasHeadOutput = true;
-      this.push(preamble);
-    }
-    this.push(
-      SitemapItem.justItem(
-        Sitemap.normalizeURL(item, this.hostname, this.lastmodDateOnly),
-        this.level
-      )
-    );
-    callback();
-  }
-
-  _flush(cb: TransformCallback): void {
-    this.push(closetag);
-    cb();
-  }
+import { Transform, TransformOptions, TransformCallback, Readable } from 'stream';
+import { ISitemapOptions } from './sitemap';
+export declare const preamble = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><urlset xmlns=\"https://www.sitemaps.org/schemas/sitemap/0.9\" xmlns:news=\"https://www.google.com/schemas/sitemap-news/0.9\" xmlns:xhtml=\"https://www.w3.org/1999/xhtml\" xmlns:image=\"https://www.google.com/schemas/sitemap-image/1.1\" xmlns:video=\"https://www.google.com/schemas/sitemap-video/1.1\">";
+export declare const closetag = "</urlset>";
+export interface ISitemapStreamOpts extends TransformOptions, Pick<ISitemapOptions, 'hostname' | 'level' | 'lastmodDateOnly'> {
+}
+export declare class SitemapStream extends Transform {
+    hostname?: string;
+    level: ErrorLevel;
+    hasHeadOutput: boolean;
+    lastmodDateOnly: boolean;
+    constructor(opts?: ISitemapStreamOpts);
+    _transform(item: ISitemapItemOptionsLoose, encoding: string, callback: TransformCallback): void;
+    _flush(cb: TransformCallback): void;
 }
-
 /**
  * Takes a stream returns a promise that resolves when stream emits finish
  * @param stream what you want wrapped in a promise
  */
-export function streamToPromise(stream: Readable): Promise<Buffer> {
-  return new Promise((resolve, reject): void => {
-    let drain: Buffer;
-    stream
-      .pipe(
-        new Writable({
-          write(chunk, enc, next): void {
-            if (!drain) {
-              drain = chunk;
-            } else {
-              drain = Buffer.concat([drain, chunk]);
-            }
-            next();
-          },
-        })
-      )
-      .on('error', reject)
-      .on('finish', () => resolve(drain));
-  });
-}
```

Comment:
why did you delete a whole bunch of code?

Reply:
lol, i am going to blame my keyboard.